cmd/gateway-gcs.go

```diff
@@ -375,19 +401,31 @@ func (l *gcsGateway) ListObjects(bucket string, prefix string, marker string, de
 		attrs, err := it.Next()
 		if err == iterator.Done {
 			break
-		} else if err != nil {
+		}
+		if err != nil {
 			return ListObjectsInfo{}, gcsToObjectError(traceError(err), bucket, prefix)
 		}
 
 		nextMarker = toGCSPageToken(attrs.Name)
 
-		if isGCSPrefix(attrs.Prefix) {
-			// we don't return our metadata prefix
+		if attrs.Prefix == gcsMinioPath {
```

Comment:
@balamurugana previously 

```
func isGCSPrefix(prefix string) bool {		
 	return strings.TrimSuffix(prefix, slashSeparator) == gcsMinioPath		
 }
```

Here isGCSPrefix() adds value as it has a crucial logic inside it `strings.TrimSuffix(prefix, slashSeparator)`

Now because of refactor it is:

```
func isGCSPrefix(prefix string) bool {		
 	return prefix == gcsMinioPath		
 }
```

This function does not add any value. i.e just for an equality statement we should not write a new function and add unit tests to it.